src/components/CategorySelect/CategorySelect.tsx

```diff
@@ -257,65 +23,34 @@ export const CategorySelect = ({
   showTooltips,
   excludeMatches = false,
   asDrawer = false,
-}: Props) => {
+}: CategorySelectProps) => {
   const { categories } = useLayerContext()
 
-  const matchOptions =
-    !excludeMatches && bankTransaction?.suggested_matches
-      ? [
-          {
-            label: 'Match',
-            options: bankTransaction.suggested_matches.map((x) => {
-              return {
-                type: OptionActionType.MATCH,
-                payload: {
-                  id: x.id,
-                  option_type: OptionActionType.MATCH,
-                  display_name: x.details.description,
-                  date: x.details.date,
-                  amount: x.details.amount,
-                  subCategories: null,
-                },
-              } satisfies CategoryOption
-            }),
-          } satisfies GroupBase<CategoryOption>,
-      ]
-      : []
+  const matches = buildMatchOptions(bankTransaction, excludeMatches)
+  const suggestions = buildSuggestedOptions(bankTransaction)
+  const allCategories = buildAllCategories(categories)
 
-  const suggestedOptions =
-    bankTransaction?.categorization_flow?.type
-    === CategorizationType.ASK_FROM_SUGGESTIONS
-      ? [
-          {
-            label: 'Suggestions',
-            options: bankTransaction.categorization_flow.suggestions.map(x =>
-              mapCategoryToOption(x),
-            ),
-          } satisfies GroupBase<CategoryOption>,
-      ]
-      : []
+  const allKeys = useMemo(() => {
+    const categoriesKeysMap = getKeysMap(allCategories.flatMap(group => group.options))
+    const matchKeysMap = getKeysMap(matches ?? [])
 
-  const categoryOptions = flattenCategories(categories)
+    return new Map([...categoriesKeysMap, ...matchKeysMap])
+  }, [allCategories, matches])
```

Comment:
ComboBox's `onSelectionChange` returns just a `Key` (for example `id`), but we need whole object in few places, ie. to properly format matches and splits. This trick stores `CategoryOption`s in map, component provides whole object in `onChange` callback.